soccer/src/soccer/coach/coach_node.cpp

```diff
@@ -0,0 +1,76 @@
+#include "coach_node.hpp"
+
+using std::placeholders::_1;
+
+CoachNode::CoachNode(const rclcpp::NodeOptions& options) : Node("coach_node", options) {
+    coach_pub_ = this->create_publisher<rj_msgs::msg::Coach>("/strategy/coach", 10);
+    playstate_change_timer_ = this->create_wall_timer(
+        100ms, std::bind(&CoachNode::check_for_playstate_change, this));
+    playstate_sub_ = this->create_subscription<rj_msgs::msg::PlayState>(
+        "/referee/play_state", 10, std::bind(&CoachNode::playstate_callback, this, _1));
+
+    current_play_state_.state = PlayState::State::Halt;
+    current_play_state_.restart = PlayState::Restart::Kickoff;
+    current_play_state_.our_restart = true;
+
+    rj_geometry_msgs::msg::Point temp_point;
+    temp_point.x = -1;
+    temp_point.y = -1;
+    current_play_state_.placement_point = temp_point;
+}
+
+void CoachNode::playstate_callback(rj_msgs::msg::PlayState::SharedPtr msg) {
+    current_play_state_ = *msg;
+    playstate_has_changed_ = true;
+}
+
+void CoachNode::check_for_playstate_change() {
+    if (playstate_has_changed_) {
+        rj_msgs::msg::Coach coach_message;
+
+        switch(current_play_state_.restart) {
+        case PlayState::Restart::Placement:
+            coach_message.match_situation = match_situation::ball_placement;
+            break;
+        case PlayState::Restart::Kickoff:
+            coach_message.match_situation = match_situation::kickoff;
+            break;
+        case PlayState::Restart::Direct:
+        case PlayState::Restart::Indirect:
+            coach_message.match_situation = match_situation::free_kick;
+            break;
+        case PlayState::Restart::Penalty:
+            coach_message.match_situation = match_situation::penalty_kick;
+            break;
+        }
+
+        if (current_play_state_.state == PlayState::State::Playing) {
+            coach_message.match_situation = match_situation::in_play;
+        }
+
+        // TODO: get possession from wherever possession is from
```

Comment:
can you make a card on [ClickUp](https://app.clickup.com/2209401/v/b/5-103744422-2) for this and then you & Anshul take it next? doesn't have to be this PR but is necessary and urgent

Reply:
recall you have to rewrite this from https://github.com/RoboJackets/robocup-software/blob/ros2/rj_gameplay/rj_gameplay/play_selector.py#L80